test/update_results_tests.jl

```diff
@@ -102,7 +102,7 @@ end
 #                           Load and analyze  DataFrame                       #
 ###############################################################################
 
-df = BSON.load(defaultname)[:df]
+df = BSON.load(defaultname)["df"]
```

Comment:
wait, why is BSON still used here, I thought we changed everything to not use BSON anymore...? I guess we missed this part. Can you please replace all BSON stuff here with JLD2? Also increase patch version in the Project.toml file?

Reply:
BSON is still a test dependency.
We only changed everything to use JLD2 _by default_.
It seems, we should have added tests, though.

Reply:
Hahah that was quite sloppy i guess. Let's use this PR as an opportunity to _replace_ BSON test dependency by JLD2

Reply:
Fair enough.

Since this is your PR @dominformant, if you're keen to get involved you could help us by replacing every usage of BSON by JLD2 in the test suite. If not, I'll probably get around to it next week since this is my fault in the first place.

Reply:
I just noticed, that [savefiles_tests.jl](https://github.com/JuliaDynamics/DrWatson.jl/blob/946400b27343afdc85c5604532ba64ed8b41f874/test/savefiles_tests.jl#L70) loops over the different file types. Shall I also remove this loop, or shouldn't we consider testing for both, `JLD2` and `BSON` as (as far as I understood) `BSON` is still supported by DrWatson, just not as default?

Reply:
Yeah you have a good point. Sorry, my memory regarding these tests is kind of vague, they were written literally years ago. Please do go forward with your proposal which indeed tests all packages, as it was initially supposed to do anyway.

Reply:
I did a first try to test for both file types, but I'm not totally happy with the result, so I will probably try to make it look nicer the next days. However, I wanted to give you the chance to take a look at the changes so that you can correct me if something is absolutely wrong.

Reply:
Looks great to me to be honest

Reply:
Okay, I added some smaller optical corrections, in my opinion this should be ready now, if you have any hints or ideas, just let me know.